scipy/integrate/quadpack.py

```diff
@@ -387,7 +387,7 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
 
     if ier in [1,2,3,4,5,7]:
         if full_output:
-            if weight in ['cos', 'sin'] and (b == Inf or a == Inf):
+            if weight in ['cos', 'sin'] and (b == Inf or a == -Inf):
```

Comment:
Looks like @ev-br agreed that this was a typo in the matching issue. Is there no easy way to catch this with a test?

Reply:
Mmm what should it actually test? I can do something like 

```
import numpy as np
from scipy import integrate

# exp(x^2)*sin(x) will never converge
def f(x):
  return np.exp(x**2)

print(len(integrate.quad(f,-np.inf,0,weight="sin",wvar=1,full_output=1)))
print(len(integrate.quad(f,0,np.inf,weight="sin",wvar=1,full_output=1)))
```
on my computer with `1.3.3` this evaluates to
```
test_quad.py:6: RuntimeWarning: overflow encountered in exp
  return np.exp(x**2)
4
5
```
[I didn't test this in the fixed version, as I do not want to download all the stuff to my local machine - but of course it might be run by github, when commited]

although I really don't like the number of return arguments to depend on (in some sense) unpredictable reasoning - see https://github.com/scipy/scipy/issues/7816

Reply:
That test works, kind of, but I wouldn't trust it to be reliable given that `== Inf` is anyway not the most robust thing to do.

Given that this fix is clearly correct, and it's just a corner case of an error message return, I'll merge it rather than risk introducing a possible flaky test.